### PR TITLE
bpo-38870: fixing unhandled hexescape in docstrings at ast.unparse

### PR DESCRIPTION
**Pull request by i64**
https://bugs.python.org/issue38870

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@CyberSaxosTiGER

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by isidentical** [comment]
CC: @pablogsal

File: Lib/ast.py [hunk 1]
```diff
@@ -1088,18 +1088,27 @@ def visit_Name(self, node):
...
     def _write_docstring(self, node):
+        def esc_char(c):
+            if c in ("\n", "\t"):
+                # In the AST form, we don't know the author's intentation
+                # about how this should be displayed. We'll only escape
+                # \n and \t, because they are more likely to be unescaped
+                # in the source
+                return c
+            else:
+                return c.encode('unicode_escape').decode('ascii')
```
Comment (pablogsal) [inline Lib/ast.py]:
```suggestion
            return c.encode('unicode_escape').decode('ascii')
```

File: Lib/test/test_unparse.py [hunk 2]
```diff
@@ -324,7 +324,11 @@ def test_docstrings(self):
             '\\t',
             '\n',
             '\\n',
-            '\r\\r\t\\t\n\\n'
+            '\r\\r\t\\t\n\\n',
+            '""">>> content = \"\"\"blabla\"\"\" <<<"""',
+            r'foo\n\x00'
+            '🐍⛎𩸽üéş^\X\BB\N{LONG RIGHTWARDS SQUIGGLE ARROW}'
```
Comment (isidentical) [inline Lib/test/test_unparse.py]:
```suggestion
            r'foo\n\x00',
            '🐍⛎𩸽üéş^\X\BB\N{LONG RIGHTWARDS SQUIGGLE ARROW}'
```

**Review by isidentical — APPROVED** [review]

**Pull request merged**